src/goto-instrument/replace_function_bodies.cpp

```diff
@@ -0,0 +1,310 @@
+/*******************************************************************\
+
+Module: Replace bodies of goto functions
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include "replace_function_bodies.h"
+
+#include <string>
+#include <utility>
+#include <memory>
+#include <util/make_unique.h>
+#include <util/message.h>
+#include <langapi/language_util.h>
+
+class replace_function_bodiest
+{
+protected:
+  virtual void replace_function_body_impl(
+    goto_functiont &function,
+    symbol_tablet &symbol_table,
+    irep_idt function_name) = 0;
+
+public:
+  virtual ~replace_function_bodiest() = default;
+
+  void replace_function_body(
+    goto_functiont &function,
+    symbol_tablet &symbol_table,
+    irep_idt function_name)
+  {
+    function.body.clear();
+    generate_parameter_names(function, symbol_table, function_name);
+    replace_function_body_impl(function, symbol_table, function_name);
```

Comment:
Should be "generate_function_body", because the body has already been removed here